entc/integration/customid/customid_test.go

```diff
@@ -116,6 +118,27 @@ func CustomID(t *testing.T, client *ent.Client) {
 	require.Equal(t, lnk.ID, blb.QueryLinks().OnlyX(ctx).ID)
 	require.Len(t, client.Blob.Query().IDsX(ctx), 3)
 
+	iSID := client.IntSID.Create().SaveX(ctx)
+	require.Equal(t, sid.ID("1"), iSID.ID)
+	iSIDChildID := sid.ID("100")
+	iSIDChild := client.IntSID.Create().SetID(iSIDChildID).SetParent(iSID).SaveX(ctx)
+	require.Equal(t, iSIDChildID, iSIDChild.ID)
+	require.Equal(t, iSID.ID, iSIDChild.QueryParent().OnlyX(ctx).ID)
+	iSIDBulk := make([]*ent.IntSIDCreate, 2)
+	iSIDBulk[0] = client.IntSID.Create().SetParent(iSID)
+	iSIDBulk[1] = client.IntSID.Create().SetParent(iSID)
+	iSIDChildren := client.IntSID.CreateBulk(iSIDBulk...).SaveX(ctx)
+	if strings.HasPrefix(t.Name(), "TestPostgres/") {
+		require.Equal(t, sid.ID("2"), iSIDChildren[0].ID)
+		require.Equal(t, sid.ID("3"), iSIDChildren[1].ID)
+	} else {
+		require.Equal(t, sid.ID("101"), iSIDChildren[0].ID)
+		require.Equal(t, sid.ID("102"), iSIDChildren[1].ID)
+	}
```

Comment:
Can you elaborate on why we have this `TestPostgres` check? Why it's different than the rest?

Reply:
First, a reason why this has exists. It checks returned values from ent whether they are set returned values from DB.

A reason of difference of code each RDBS is below.

Mysql and sqlite assign the next value which is from maximum value of the column when auto increment.

> https://dev.mysql.com/doc/refman/8.0/en/example-auto-increment.html
> "When you insert any other value into an AUTO_INCREMENT column, the column is set to that value and the sequence is reset so that the next automatically generated value follows sequentially from the largest column value. "
> https://www.sqlite.org/autoinc.html
> "The ROWID chosen for the new row is at least one larger than the largest ROWID"
> 

Auto-increment of Postgres is represented as a type named Serial. ( off course, you know. ) And it behaves as a function `nextval（' tablename_colname_seq'）`
https://www.postgresql.org/docs/13/datatype-numeric.html#DATATYPE-SERIAL
https://www.postgresql.org/docs/13/functions-sequence.html

And ent creates a table by SQL below when test of IntSID run.
```sql
CREATE TABLE "int_si_ds" (
  "id" bigint NOT NULL GENERATED BY DEFAULT AS IDENTITY,
  "int_sid_parent" bigint NULL, PRIMARY KEY ("id"),
  CONSTRAINT "int_si_ds_int_si_ds_parent" FOREIGN KEY ("int_sid_parent") REFERENCES "int_si_ds" ("id") ON DELETE SET NULL
)
```

> https://www.geeksforgeeks.org/postgresql-identity-column/#:~:text=The%20GENERATED%20ALWAYS%20instructs%20PostgreSQL,value%20for%20the%20identity%20column.
> "The GENERATED AS IDENTITY constraint is the SQL standard-conforming variant of the PostgreSQL’s SERIAL column."

Unfortunately, Serial will not associate with column value user inserted.

Therefore, SaveX() will not return "101" or "102" as ID after insertion of sid.ID("100") on Postgres.

Even worse, Postgres will occer an error if insert ID "2" instead of "100". Because it will conflict between sequence and exsisting value.

```text
panic: ent: constraint failed: insert nodes to table "int_si_ds": pq: duplicate key value violates unique constraint "int_si_ds_pkey"
```

Therefore, I cannot unify the test Postgres and the others.

Reply:
I unified the tests by a way of avove.
Because, fundamental purpose is just checking children IDs. This if-statement is noisy.
https://github.com/ent/ent/pull/2657/commits/94163af0a4d9378d2a6cf7e224e485f5f9975fb9